node/src/accountant/scanners/payable_scanner/mod.rs

```
@@ -0,0 +1,687 @@
+                }
+
+                NextScanToRun::PendingPayableScan
+            }
+            Err(_e) => match msg.payable_scan_type {
+                PayableScanType::New => NextScanToRun::NewPayableScan,
+                PayableScanType::Retry => NextScanToRun::RetryPayableScan,
+            },
+        }
+    }
+
+    fn process_message(&self, msg: &SentPayables, logger: &Logger) {
+        match &msg.payment_procedure_result {
+            Ok(batch_results) => match msg.payable_scan_type {
+                PayableScanType::New => self.handle_new(batch_results, logger),
+                PayableScanType::Retry => self.handle_retry(batch_results, logger),
+            },
+            Err(local_error) => Self::log_local_error(local_error, logger),
+        }
+    }
+
+    fn handle_new(&self, batch_results: &BatchResults, logger: &Logger) {
+        let (sent, failed) = calculate_lengths(&batch_results);
+        debug!(
+            logger,
+            "Processed new txs while sending to RPC: {}",
+            batch_stats(sent, failed),
+        );
+        if sent > 0 {
+            self.insert_records_in_sent_payables(&batch_results.sent_txs);
+        }
+        if failed > 0 {
+            self.insert_records_in_failed_payables(&batch_results.failed_txs);
+        }
+    }
+
+    fn handle_retry(&self, batch_results: &BatchResults, logger: &Logger) {
+        let (sent, failed) = calculate_lengths(&batch_results);
+        debug!(
+            logger,
+            "Processed retried txs while sending to RPC: {}",
+            batch_stats(sent, failed),
+        );
+
+        if sent > 0 {
+            self.insert_records_in_sent_payables(&batch_results.sent_txs);
+            self.update_statuses_of_prev_txs(&batch_results.sent_txs);
+        }
+        if failed > 0 {
+            // TODO: Would it be a good ides to update Retry attempt of previous tx?
+            Self::log_failed_txs_during_retry(&batch_results.failed_txs, logger);
```

Ironically, you did it perfectly the other way around. Here, for the failures, we actually do want to act instead of just logging, on contrary to the successes where we want to do nothing at all.  

The desired operations are like this:

1) Insert new records into failed_payable, those should correspond to these attempts that have failed.
2) Delete those records from sent_payable that match these failed attempts. (Probably by the wallet address). 

Ok?

---

The reason we are logging is that these payables will eventually be retried based on the previous failed records. By logging them, we can prevent inserting the same payables again. Don't worry about it; I've thought this through.

This TODO is about updating the previous failed payable records with the data from the next failed transaction. However, this update is not necessary.